io/src/read_detector_description.cpp

```diff
@@ -0,0 +1,175 @@
+                                       geometry_file.data());
+
+    // Fill the detector description with information about the (sensitive)
+    // surfaces, and the digitization configurations belonging to those
+    // surfaces.
+    dd.reserve(surfaces.size());
+    for (const auto& [geom_id, transform] : surfaces) {
+
+        // Acts geometry identifier for the surface.
+        const Acts::GeometryIdentifier acts_geom_id{geom_id};
+
+        // Skip non-sensitive surfaces. They are not needed in the
+        // "detector description".
+        if (acts_geom_id.sensitive() == 0) {
+            continue;
+        }
+
+        // Add a new element to the detector description.
+        dd.resize(dd.size() + 1);
+
+        // Fill the new element with the geometry ID and the transformation of
+        // the surface in question.
+        dd.surface_link().back() = detray::geometry::barcode{geom_id};
+        dd.placement().back() = transform;
+        dd.geometry_id().back() = geom_id;
+
+        // Find the module's digitization configuration.
+        const traccc::digitization_config::Iterator digi_it =
+            digi.find(acts_geom_id);
+        if (digi_it == digi.end()) {
+            std::ostringstream msg;
+            msg << "Could not find digitization config for geometry ID: "
+                << acts_geom_id;
+            throw std::runtime_error(msg.str());
+        }
+
+        // Fill the new element with the digitization configuration for the
+        // surface.
+        fill_digi_info(dd, *digi_it);
+    }
+}
+
+void read_json_dd(traccc::silicon_detector_description::host& dd,
+                  std::string_view geometry_file,
+                  const traccc::digitization_config& digi) {
+
+    // Construct a (temporary) Detray detector object from the geometry
+    // configuration file.
+    vecmem::host_memory_resource mr;
+    traccc::default_detector::host detector{mr};
+    const traccc::default_detector::host::geometry_context ctx{};
+    traccc::io::read_detector(detector, mr, geometry_file);
+
+    // Iterate over the surfaces of the detector.
+    const traccc::default_detector::host::surface_lookup_container& surfaces =
+        detector.surfaces();
+    dd.reserve(surfaces.size());
+    for (const auto& surface_desc : detector.surfaces()) {
+
+        // Acts geometry identifier(s) for the surface.
+        const traccc::geometry_id geom_id{surface_desc.source};
+        const Acts::GeometryIdentifier acts_geom_id{geom_id};
+
+        // Skip non-sensitive surfaces. They are not needed in the
+        // "detector description".
+        if (acts_geom_id.sensitive() == 0) {
+            continue;
+        }
+
+        // Add a new element to the detector description.
+        dd.resize(dd.size() + 1);
+
+        // Construct a Detray surface object.
+        const detray::tracking_surface<traccc::default_detector::host> surface{
+            detector, surface_desc.barcode()};
```

Comment:
If we already have the correct descriptor, no need to fetch it from the barcode again
```suggestion
            detector, surface_desc};
```

Reply:
Let me give that a try. I remember the compiler being unhappy here with some of the things that I tried...

Reply:
:+1: